src/scheduler/analysis.py

```diff
@@ -229,18 +229,19 @@ def get_plugin_dict(self) -> dict:
         '''
         plugin_list = self._get_list_of_available_plugins()
         plugin_list = self._remove_unwanted_plugins(plugin_list)
-        default_plugins = self._get_default_plugins_from_config()
-        default_flag_dict = {}
+        plugin_sets = self._get_plugin_sets_from_config()
         result = {}
         for plugin in plugin_list:
+            current_plugin_plugin_sets = {}
             mandatory_flag = plugin in MANDATORY_PLUGINS
-            for key in default_plugins:
-                default_flag_dict[key] = plugin in default_plugins[key]
+            for plugin_set in plugin_sets:
+                current_plugin_plugin_sets[plugin_set] = plugin in plugin_sets[plugin_set]
             blacklist, whitelist = self._get_blacklist_and_whitelist_from_plugin(plugin)
+            # TODO this should not be a tuple but rather a dictionary/class
```

Comment:
There was a reason for this once. But I forgot it ...

Reply:
Then I'd prefer to have the comment.